lib/rubocop_director/rubocop_stats.rb

```diff
@@ -7,22 +7,48 @@
 module RubocopDirector
   class RubocopStats
     include Dry::Monads[:result]
+    include Dry::Monads::Do.for(:fetch)
+
+    TEMP_CONFIG_PATH = "./.temp_rubocop.yml"
 
     def fetch
-      _, stderr = Open3.capture3("sed '/todo/d' ./.rubocop.yml > tmpfile; mv tmpfile ./.rubocop.yml")
-      if stderr.length > 0
-        return Failure("Failed to remove TODO from rubocop config: #{stderr}")
-      end
+      config = yield load_config
+      yield temp_config(initial_config: config)
+
+      stats = yield generate_stats
+      yield remove_temp_config
+
+      Success(stats)
+    end
+
+    private
 
-      stdout, stderr = Open3.capture3("bundle exec rubocop --format json")
+    def load_config
+      Success(YAML.load_file("./.rubocop.yml"))
+    rescue
+      Failure("unable to load rubocop config. Please ensure .rubocop.yml file is present at your project's root directory")
+    end
+
+    def temp_config(initial_config:)
```

Comment:
I'd rather call it something like `generate_temp_rubocop_config_without_todo` to make a clear intention

Reply:
makes sense